### PR TITLE
REACH-148 NodeView for PythonNode

### PR DESCRIPTION
**Pull request by pbidenko**
Implemented NodeView for PythonNode.
Created base node for nodes with variable inputs.
To work correctly, this pull request requires https://github.com/Benglin/Dynamo/pull/269

### CODE REVIEW

File: app/app.html [hunk 1]
```diff
@@ -862,6 +866,25 @@ <h3 id="feedback-title">
 
 </script>
 
+<script type="text/template" id="python-template">
```
Comment (pbidenko) [inline app/app.html]: Template for Python editor window.

File: app/scripts/models/nodes/PythonScriptNode.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+define(['VariableInputNode', 'FLOOD'], function (VariableInputNode, FLOOD) {
```
Comment (pbidenko) [inline app/scripts/models/nodes/PythonScriptNode.js]: Initialize with some simple script, just like in Dynamo.

File: app/scripts/models/responses/NodeCreationDataResponse.js [hunk 3]
```diff
@@ -24,6 +24,10 @@ define(['backbone', 'staticHelpers'], function (Backbone, staticHelpers) {
                 returnValue.extra.code = data.Code;
             }
 
+            if (data.Script) {
```
Comment (pbidenko) [inline app/scripts/models/responses/NodeCreationDataResponse.js]: Script received from Dynamo after Dyn file opening.

File: app/scripts/views/AppView.js [hunk 4]
```diff
@@ -245,6 +248,16 @@ define([  'backbone',
       }
     },
 
+    viewPythonEditor: function(e){
```
Comment (pbidenko) [inline app/scripts/views/AppView.js]: Open PythonScript editor view.

**Comment by pbidenko** [comment]
Hi @pboyer,

I fixed conflict. Also Ben merged corresponding PR in Dynamo.
Please, take a look.

**Comment by pboyer** [comment]
Hi @pbidenko,

A few things:

1)  We shouldn't be exposing specific node implementation details on the `AppView`.  It's exposed appropriately as an event, but what we need here is something a bit more general.  It seems like we could more than one stone by introducing a single element that handles modal contexts like this.  Consider calling it something like `ModalView`, then the `PythonEditorView` can be created in the node and then injected into this element.  Does that make sense?

2)  In attempting to use this code, it didn't really work.  The node can't add/remove ports, nor can I see the editor.  Maybe there was a mistake in the way I merged it in.

~Peter

**Comment by pbidenko** [comment]
Hi @pboyer, please take a look.

**Pull request closed without merging**